src/main/java/de/tum/in/www1/artemis/service/exam/ExamDateService.java

```diff
@@ -35,6 +36,16 @@ public ExamDateService(ExamRepository examRepository, StudentExamRepository stud
         this.studentExamRepository = studentExamRepository;
     }
 
+    /**
+     * Returns if the exam is over by checking if the exam end date has passed.
+     *
+     * @param exam the exam
+     * @return true if the exam is over
+     */
+    public boolean isExamOver(Exam exam) {
+        return exam.getEndDate().isBefore(ZonedDateTime.now());
```

Comment:
At early stages the dates might not be set in an exam yet. How should this method handle these cases?

Reply:
end date is always defined:
```
@NotNull
    public ZonedDateTime getEndDate() {
        return endDate;
    }
```